### PR TITLE
MVP

### LINKED ISSUE #1: MVP
Send events to Temporal Server task queue

There are two events
- Event
- Admin Event

Each event type will have it's own interface for activity
- KeycloakEventWorkflow
- KeycloakAdminEventWorkflow

each of the interfaces will have a send method and event parameter

some configuration options
- server (hostname)(string) defaults to "localhost:7233"
- namespace (string) defaults to "default"
- queue (string) defaults to "keycloak"

### LINKED ISSUE #4: Add caching to pipleline for dependencies
See https://github.com/actions/setup-java?tab=readme-ov-file#caching-packages-dependencies

### PR DESCRIPTION
**Pull request by snovak7**
Closes #1
Closes #4

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+      - name: PR Check
+        if: github.event_name == 'pull_request'
+        run: mvn --batch-mode compile
+      - name: Publish package
+        run: mvn --batch-mode deploy
+        if: github.event_name != 'pull_request'
```
Comment (ElanHasson) [inline .github/workflows/release.yml:31]: how does release on a tag work for this?

File: src/main/kotlin/app/infinityflow/keycloak/workflows/admin/KeycloakAdminEventWorkflow.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+import app.infinityflow.keycloak.model.admin.Event
+import io.temporal.workflow.WorkflowInterface
+import io.temporal.workflow.WorkflowMethod
+
+@WorkflowInterface
+interface KeycloakAdminEventWorkflow {
+    @WorkflowMethod
+    fun send(event: Event)
+}
```
Comment (ElanHasson) [inline src/main/kotlin/app/infinityflow/keycloak/workflows/admin/KeycloakAdminEventWorkflow.kt:11]:
```suggestion
}

```

File: src/test/kotlin/app/infinityflow/keycloak/events/tests/TemporalTest.kt [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import org.keycloak.events.EventType
+import org.keycloak.events.admin.AdminEvent
+import org.keycloak.events.admin.AuthDetails
+import org.keycloak.events.admin.OperationType
+import org.keycloak.events.admin.ResourceType
+import java.util.Date
+import java.util.UUID
+import kotlin.test.Test
+
+class TemporalTest {
```
Comment (ElanHasson) [inline src/test/kotlin/app/infinityflow/keycloak/events/tests/TemporalTest.kt:14]: i didn't see anything about tests in the pipeline. Will these be run?

**Comment by ElanHasson** [comment]
LGTM-- never worked in Java, but this seems reasonable.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
+name: Release
+on:
+  push:
+    branches:
+      - main
+  pull_request: {}
+  release:
+    types:
+      - published
+jobs:
```
Comment (ElanHasson) [inline .github/workflows/release.yml:10]: I added https://github.com/InfinityFlowApp/keycloak-event-listener-temporal/issues/4 to setup dependency caching to speed up builds.

File: .github/workflows/release.yml (hunk 1 above)
Comment (snovak7) [reply .github/workflows/release.yml:31]: Maven extension puts a version on pom.xml check .mvn folder

File: src/test/kotlin/app/infinityflow/keycloak/events/tests/TemporalTest.kt (hunk 3 above)
Comment (snovak7) [reply src/test/kotlin/app/infinityflow/keycloak/events/tests/TemporalTest.kt:14]: In general it should as maven runs them in the build lifecycle

If you want to skip tests you would have to run -DskipTests=true

**Review by ElanHasson — APPROVED** [review]

**Pull request merged**